src/raw-value.ts

```diff
@@ -239,6 +262,78 @@ export function reverseProxyFactory(env: MembraneBroker) {
             }
             return env.getRawValue(sec);
         }
+        has(shadowTarget: ReverseShadowTarget, key: PropertyKey): boolean {
+            const { target } = this;
+            // TODO [#60]: protect against leaking symbols
+            return key in target;
+        }
+        ownKeys(shadowTarget: ReverseShadowTarget): PropertyKey[] {
+            const { target } = this;
+            // TODO [#60]: protect against leaking symbols
+            return ownKeys(target);
+        }
+        isExtensible(shadowTarget: ReverseShadowTarget): boolean {
+            if (!ReflectIsExtensible(shadowTarget)) {
+                return false;
+            }
+            const { target } = this;
+            if (!ReflectIsExtensible(target)) {
+                lockShadowTarget(shadowTarget, target);
+                return false;
+            }
+            return true;
+        }
+        getOwnPropertyDescriptor(shadowTarget: ReverseShadowTarget, key: PropertyKey): PropertyDescriptor | undefined {
+            const { target } = this;
+            // TODO [#60]: leaking key as symbol
+            let rawDesc = ReflectGetOwnPropertyDescriptor(shadowTarget, key);
+            if (!isUndefined(rawDesc)) {
+                return rawDesc;
+            }
+            const secDesc = ReflectGetOwnPropertyDescriptor(target, key);
+            if (isUndefined(secDesc)) {
+                return secDesc;
+            }
+            rawDesc = getRawDescriptor(secDesc);
+            if (secDesc.configurable == false) {
+                // updating the descriptor to non-configurable on the shadow
+                ReflectDefineProperty(shadowTarget, key, rawDesc);
+            }
+            return rawDesc;
+        }
+        getPrototypeOf(shadowTarget: ReverseShadowTarget): RawValue {
+            const { target } = this;
+            return env.getRawValue(ReflectGetPrototypeOf(target));
+        }
+        setPrototypeOf(shadowTarget: ReverseShadowTarget, prototype: RawValue): boolean {
+            const { target } = this;
+            return ReflectSetPrototypeOf(target, env.getSecureValue(prototype));
+        }
+        preventExtensions(shadowTarget: ReverseShadowTarget): boolean {
+            if (ReflectIsExtensible(shadowTarget)) {
+                const { target } = this;
+                if (!ReflectPreventExtensions(target)) {
+                    // if the target is a proxy manually created in the sandbox, it might reject
+                    // the preventExtension call, in which case we should not attempt to lock down
+                    // the shadow target.
+                    return false;
```

Comment:
In the comment above how do we know the `!ReflectPreventExtensions(target)` is a proxy? Also since `lockShadowTarget` only locks the `shadowTarget` wouldn't it avoid throwing on the `target` since we're locking the `shadowTarget` and not the `target` down?

Reply:
hmm, I'm not sure I understand the issue here. The language invariants are applied here, so if it says that the target is not extensible, it should not be extensible by any means.

Reply:
```
var o = new Proxy({}, { preventExtension() { return true }, isExtensible() { return true } });
// undefined
Reflect.isExtensible(o)
// true
Reflect.preventExtensions(o)
// true
Reflect.isExtensible(o)
VM415:1 Uncaught TypeError: 'isExtensible' on proxy: trap result does not reflect extensibility of proxy target (which is 'false')
    at Object.isExtensible (<anonymous>)
    at <anonymous>:1:9
(anonymous) @ VM415:1
```

What this means is that the only case in which we can lockShadowTarget() is when we are sure that the target is known to be non-extensible.